test/unit/test_utils.py

```diff
@@ -55,6 +55,19 @@ def test_copy_touched_file(dest_file, source_file):
     assert not copy_file(source_file, dest_file)
 
 
+def test_copy_file_with_destination_directory(dest_file, source_file):
+    # Change source file to trigger copy_file
+    source_file.write_text('foo\nbar\nzoo')
+
+    with pytest.raises(Exception):
+        copy_file(source_file, '/tmp')
+
+    with pytest.raises(Exception):
+        copy_file('/tmp', dest_file)
+
```

Comment:
Looks great! But since Exception is pretty broad, let's add an assert to make sure we are getting the Exception message we expect. If you capture the exception in a var (say, `error`), you can check the message, like:

```
assert "can not be a directory" in str(error.value.args[0])
```